ComputeDemo/ComputeAlphaMainIntegrationTest.cs

```diff
@@ -0,0 +1,115 @@
+
+            _output.WriteLine($"Retrieving address with the name: {addrName}");
+
+            var exc = xunit.Assert.ThrowsAny<grpccore.RpcException>(
+                () => addressesClient.Get(_project, _region, addrName)
+            );
+            xunit.Assert.Equal(grpccore.StatusCode.NotFound, exc.StatusCode);
+
+            Address addressResource = new Address
+            {
+                Name = addrName,
+                Region = _region,
+                NetworkTier = Address.Types.NetworkTier.Premium
+            };
+
+            _output.WriteLine($"Creating address with the name: {addrName}");
+            Operation insertOp = addressesClient.Insert(_project, _region, addressResource);
+            _output.WriteLine($"Operation to create address: {insertOp.Name} status {insertOp.Status}; start time {insertOp.StartTime}");
+            insertOp = PollForCompletion(insertOp, "create");
+            _output.WriteLine($"Operation to create address completed: status {insertOp.Status}; start time {insertOp.StartTime}; end time {insertOp.EndTime}");
+
+            _output.WriteLine($"Retrieving address with the name: {addrName}");
+            Address readAddr = addressesClient.Get(_project, _region, addrName);
+            xunit.Assert.NotNull(readAddr);
+            xunit.Assert.Equal(readAddr.Name, addrName);
+
+            _output.WriteLine($"Deleting address with the name: {addrName}");
+            Operation deleteOp = addressesClient.Delete(_project, _region, addrName);
+            _output.WriteLine($"Operation to delete address: {deleteOp.Name} status {deleteOp.Status}; start time {deleteOp.StartTime}");
+            deleteOp = PollForCompletion(deleteOp, "delete");
+            _output.WriteLine($"Operation to delete address completed: status {deleteOp.Status}; start time {deleteOp.StartTime}; end time {deleteOp.EndTime}");
+        }
+
+        private Operation PollForCompletion(Operation insertOp, string alias)
+        {
+            RegionOperationsClient regionOperationsClient = _ropsClient;
+
+            var localOps = insertOp;
+
+            TimeSpan timeOut = TimeSpan.FromMinutes(3);
+            TimeSpan pollInterval = TimeSpan.FromSeconds(15);
+            DateTime pollStartTime = DateTime.UtcNow;
+
+            while (localOps.Status != Operation.Types.Status.Done)
+            {
+                GetRegionOperationRequest request = new GetRegionOperationRequest
+                {
+                    Operation = insertOp.Name,
+                    Region = _region,
+                    Project = _project,
+                };
+                _output.WriteLine($"Checking for {alias} operation status ...");
+                localOps = regionOperationsClient.Get(request);
+                
+
+                if (localOps.Status == Operation.Types.Status.Done) 
+                {
+                    break;
+                }
+                if((DateTime.UtcNow - pollStartTime) > timeOut)
```

Comment:
I'd suggest computing a deadline at the start of the method (`DateTime.UtcNow + timeout`) and then you can just check `if (DateTime.UtcNow > deadline)`

Reply:
👍